x/ibc-hooks/contracts/crosschain-swaps/src/execute.rs

```diff
@@ -0,0 +1,271 @@
+            error: format!("this should be unreachable"),
+            memo: memo.to_string(),
+        })?
+    } else {
+        serde_json_wasm::from_str("{}").unwrap()
+    };
+
+    // If tracking callbacks, we want to include the callback key in the memo
+    // without otherwise modifying the provided one
+    let memo = match config.track_ibc_callbacks {
+        true => {
+            let serde_cw_value::Value::Map(mut m) = memo else { unreachable!() };
+            m.insert(
+                serde_cw_value::Value::String(CALLBACK_KEY.to_string()),
+                serde_cw_value::Value::String(contract_addr.to_string()),
+            );
+            serde_cw_value::Value::Map(m)
+        }
+        false => memo,
+    };
+
+    // Serialize the memo. If it is an empty json object, set it to ""
+    let mut memo_str =
+        serde_json_wasm::to_string(&memo).map_err(|_e| ContractError::InvalidMemo {
+            error: "could not serialize".to_string(),
+            memo: format!("{:?}", swap_msg_state.forward_to.next_memo),
+        })?;
+    if memo_str == "{}" {
+        memo_str = String::new();
+    }
+
+    // Cosmwasm's  IBCMsg::Transfer  does not support memo.
+    // To build and send the packet properly, we need to send it using stargate messages.
+    // See https://github.com/CosmWasm/cosmwasm/issues/1477
+    let ibc_transfer = MsgTransfer {
+        source_port: "transfer".to_string(),
+        source_channel: swap_msg_state.forward_to.channel.clone(),
+        token: Some(
+            Coin::new(
+                swap_response.amount.into(),
+                swap_response.token_out_denom.clone(),
+            )
+            .into(),
+        ),
+        sender: contract_addr.to_string(),
+        receiver: swap_msg_state.forward_to.receiver.clone().into(),
+        timeout_height: None,
+        timeout_timestamp: Some(ts.nanos()),
+        memo: memo_str,
+    };
+
+    // Base response
+    let response = Response::new()
+        .add_attribute("status", "ibc_message_created")
+        .add_attribute("ibc_message", format!("{:?}", ibc_transfer));
+
+    if !config.track_ibc_callbacks || swap_msg_state.forward_to.failed_delivery.is_none() {
+        // If we're not tracking callbacks, or there isn't any recovery addres,
+        // then there's no need to listen to the response of the send.
+        // The response data
+
+        // Add the response data since it won't be added in the reply.
+        let amount = swap_response.amount;
+        let denom = swap_response.token_out_denom;
+        let channel_id = swap_msg_state.forward_to.channel;
+        let to_address = swap_msg_state.forward_to.receiver;
+        let data = CrosschainSwapResponse {
+            msg: format!("Sent {amount}{denom} to {channel_id}/{to_address}"),
+        };
+
+        return Ok(response
+            .set_data(to_binary(&data)?)
+            .add_message(ibc_transfer));
+    }
```

Comment:
ngl, don't really understand why the data format for the response is the way it is.

Reply:
or what the add message is here